doc/specs/stdlib_logger.md

```diff
@@ -0,0 +1,934 @@
+* precede messages by a blank line;
+* precede messages by a time stamp of the form
+  `yyyy-mm-dd hh:mm:ss.sss`;
+* precede messages with the names of a module and procedure;
+* follow a message with the `stat` and `errmsg` of the error report
+  that prompted the log message;
+* follow a message with the `iostat` and `iomsg` of the I/O error
+  report that prompted the log message;
+* label a message with one of `'INFORMATION: '`, `'WARNING: '`,
+  `'ERROR: '`, or `'I/O ERROR: '`;
+* indent subsequent lines of the messages; and
+* format the text to fit within a maximum column width.
+
+Note: Loggers of type `logger_t` normally report their messages to I/O
+units in the internal list termed `log_units`. However if `log_units`
+is empty then the messages go to the `output_unit` of the intrinsic
+module `iso_fortran_env`.
+
+
+## The `STDLIB_LOGGER` constants
+
+The module defines nine distinct public integer constants for
+reporting errors in the `stat` arguments of some of the module's
+procedures. The constants, termed error codes, are as follows:
+
+| Error Code | Description |
+|------------|-------------|
+| `success` | no error was detected|
+| `close_failure` | a `close` statement for an I/O unit failed|
+| `invalid_index_error` | the `column` was invalid for the given `line` |
+| `non_sequential_error` | the I/O unit did not have `SEQUENTIAL` access|
+|`open_failure` | an `open` statement failed |
+| `read_only_error` | an output unit did not have an `access` specifier of `'WRITE'` or `'READWRITE'` |
+| `unformatted_in_error` | the unit did not have a `form` of `'FORMATTED'`|
+|`unopened_in_error`| the unit was not opened|
+| `write_fault` | one of the writes to `log_units` failed|
+
+## The derived type: logger_t
+
+### Status
+
+Experimental
+
+### Description
+
+Serves to define 'logger' variables to be used in reporting
+significant events encountered during the execution of a program.
+
+### Syntax
+
+type(logger_t) :: variable
+
+### Private attributes
+
+| Attribute | Type | Description | Initial value |
+|-----------|------|-------------|---------| 
+| `add_line` | Logical | Flag to precede output with a blank line |`.true.`|
+|`indent_lines` | Logical| Flag to indent subsequent lines by four columns|`.true.`|
+|`log_units`|Integer array| List of I/O units used for output| empty|
+|`max_width`| Integer | Maximum column width of output | 80 |
+|`time_stamp`| Logical | Flag to precede output by a time stamp | `.true.` |
+|`units` | Integer | Count of the number of active output units | 0 |
+
+## The `stdlib_logger` variable
+
+The module defines one public variable, `global_logger`, of type
+`logger_t`. As might be guessed from its name, `global_logger` is
+intended to serve as the default logger for use throughout an
+application.
+
+
+### Overview of the `logger_t` methods
+
+The module defines twelve public procedures: one function and eleven
+subroutines. All are methods of the `logger_t` derived type. The
```

Comment:
```suggestion
subroutines.  The
```
the sentence "All are methods of the `logger_t` derived type." is not needed if this section is part of "derived type: logger_t"